### PR TITLE
Adds two buttons to the admin traitor panel

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Adds "Open VV" to the admin traitor panel, which opens that antagonist's VV panel.
- The "Mark as in/complete" button in the traitor panel now updates color based on whether the objective is completed, instead of whether an admin has marked it to be complete. 
  - In most cases, it'll end up looking the same / similar as it does currently.
  - If the objective is complete without admin intervention, it'll also show up as green
  - Some objectives disregard the admin completion value, in which case it will not change color, only text

## Why It's Good For The Game

- More complex antags, like lings or heretics, often require that admins view the datum's variables to debug, This makes it easier / faster to locate them. 
- People will often ask if their objective is complete or not, and heretics straight up require complete objectives. This makes it easier for admins to see which are complete or not by the code's standards. 

## Changelog

:cl: Melbert
admin: Adds "Open VV" to traitor panel antags, which opens that antag's vv panel.
admin: "Mark as in/completed" by objectives in the traitor panel now changes color based on whether the objective is determined to be "complete", instead of whether an admin has marked it to be complete. For some objectives, it might not update color (only text) when an admin marks it as completed - this means the objective disregards whether an admin has marked it as complete. 
/:cl:

### CODE REVIEW

File: code/modules/admin/antag_panel.dm [hunk 1]
```diff
@@ -46,7 +46,12 @@ GLOBAL_VAR(antag_prototypes)
 	else
 		var/obj_count = 1
 		for(var/datum/objective/objective as anything in objectives)
-			result += "<B>[obj_count]</B>: [objective.explanation_text] <a href='?src=[REF(owner)];obj_edit=[REF(objective)]'>Edit</a> <a href='?src=[REF(owner)];obj_delete=[REF(objective)]'>Delete</a> <a href='?src=[REF(owner)];obj_completed=[REF(objective)]'><font color=[objective.completed ? "green" : "red"]>[objective.completed ? "Mark as incomplete" : "Mark as complete"]</font></a><br>"
+			result += "<B>[obj_count]</B>: [objective.explanation_text] \
+				<a href='?src=[REF(owner)];obj_edit=[REF(objective)]'>Edit</a> \
+				<a href='?src=[REF(owner)];obj_delete=[REF(objective)]'>Delete</a> \
+				<a href='?src=[REF(owner)];obj_completed=[REF(objective)]'><font color=[objective.completed ? "green" : "red"]>[objective.completed ? "Mark as incomplete" : "Mark as complete"]</font></a> \
+				<a href='?src=[REF(owner)];is_obj_completed=[REF(objective)]'>Check completion</a> \
```
Comment (Ryll-Ryll) [inline code/modules/admin/antag_panel.dm:53]:
```suggestion
				<a href='?src=[REF(owner)];is_obj_completed=[REF(objective)]'>Check Completion</a> \
```consistency

**Review by Ryll-Ryll — APPROVED** [review]
Very good, this has been something we could've used for a while now

**Comment by optimumtact** [comment]
Why do you need to check completion? the colour of the complete button tells you if it's complete or not?

**Comment by Ryll-Ryll** [comment]
> Why do you need to check completion? the colour of the complete button tells you if it's complete or not?

That's for forcing completion of the objective, it doesn't reflect completion status

**Comment by optimumtact** [comment]
> > Why do you need to check completion? the colour of the complete button tells you if it's complete or not?
> 
> That's for forcing completion of the objective, it doesn't reflect completion status

I thought it was green if the objective was complete, and red if it wasn't

**Comment by Jordie0608** [comment]
It's a slightly different check, `objective.completed` instead of `objective.check_completion()` but would probably evaluate to the same result in most cases:
https://github.com/tgstation/tgstation/blob/12d01991279c0685911e65a0c75b862b3fda1410/code/modules/admin/antag_panel.dm#L52
Just make this use `objective.check_completion()`?

**Comment by MrMelbert** [comment]
Right now it shows up as green if an admin's forced completion of the objective, or red if an admin has not (tracked by `objective.completed`)
Not all objectives even bother checking `completed` in `check_completion()` - but I can swap it over anyways

**Pull request merged**